acmw-db-app/pages/api/attendance/[pid].js

```diff
@@ -0,0 +1,64 @@
+import pgQuery from '../../../postgres/pg-query.js';
+//idk I wanna do this somehow but need to export it from student? thots?
+//import {createStudent} from '../student/[pid].js'; 
+
+// POST /api/attendance/record
+// Record a specific student attendance
+// event_code: code for meeting, f_name: first name of student, 
+// l_name_dot_num: last name and dot number of student, year_level: school level of student
+async function record (event_code, f_name, l_name_dot_num, year_level) {
+    const curr_student_id = await pgQuery(`SELECT id FROM student WHERE name_dot_num = '${l_name_dot_num}'`);
+    //if the student doesn't exist 
+    if(!curr_student_id.rowCount){
+       let l_name = l_name_dot_num.split(/\./)[0];
+       // creating new student if it didn't exist
+       // so, right now (1/3/2021) should get
+       //  error: insert or update on table "meeting_student" violates foreign key constraint "meeting_student_meeting_id_fkey"
+       curr_student_id = await pgQuery(`
+            INSERT INTO student (fname, lname, name_dot_num, school_level)
+                VALUES ('${f_name}', '${l_name}', '${l_name_dot_num}',
+                 '${year_level}')
+         RETURNING id;
+         `);
+    }
+
+    // assuming meeting id is present for now, so not adding it to meeting table, just straight to meeting student
+    const data = await pgQuery(`INSERT INTO meeting_student (meeting_id, student_id)
+        VALUES ('${event_code}', '${curr_student_id.rows[0].id}')`);
```

Comment:
meeting_id and event code are not the same thing, meeting_id should be the actual id from the meeting table corresponding to the meeting with matching event code

Reply:
so there needs to be another SELECT query sometime before this

Reply:
oopsies that's my b